lib/plsql/procedure_call.rb

```diff
@@ -236,6 +236,14 @@ def add_argument(argument, value, argument_metadata=nil)
         @bind_values[argument] = value.nil? ? nil : (value ? 1 : 0)
         @bind_metadata[argument] = argument_metadata.merge(:data_type => "NUMBER", :data_precision => 1)
         "l_#{argument}"
+      when 'UNDEFINED'
+        if argument_metadata[:type_name] == 'XMLTYPE'
+          @declare_sql << "l_#{argument} XMLTYPE;\n"
+          @assignment_sql << "l_#{argument} := XMLTYPE(:#{argument});\n" if not value.nil?
+          @bind_values[argument] = value if not value.nil?
+          @bind_metadata[argument] = argument_metadata.merge(:data_type => "CLOB")
+          "l_#{argument}"
+        end
```

Comment:
I assume that all the existing functionality is covered by unit tests and all existing unit tests are passing.
I did not verify what other types fall under UNDEFINED, but for sure none of them is currently supported by framework.